app/src/main/java/co/co/nimblehq/showcases/poly/place/ui/screens/restaurant/uimodel/RestaurantUiModel.kt

```diff
@@ -0,0 +1,46 @@
+package co.co.nimblehq.showcases.poly.place.ui.screens.restaurant.uimodel
+
+import android.graphics.Bitmap
+import android.location.Location
+import android.os.Parcelable
+import co.co.nimblehq.showcases.poly.place.domain.model.Restaurant
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+data class RestaurantUiModel(
+    val id: String,
+    val thumbnailImage: Bitmap?,
+    val name: String,
+    val address: String,
+    val distance: Float?
+) : Parcelable
+
+fun Restaurant.toRestaurantUiModel(currentLatitude: Double, currentLongitude: Double) =
+    RestaurantUiModel(
+        id = id,
+        thumbnailImage = null,
+        name = name,
+        address = address,
+        distance = if (latitude != null && longitude != null) {
+            calculateDistance(
+                startLat = currentLatitude,
+                startLng = currentLongitude,
+                endLat = latitude!!,
+                endLng = longitude!!
```

Comment:
Can we avoid using this `!!`?
```suggestion
                endLat = latitude ?: 0.0,
                endLng = longitude ?: 0.0
```